packages/qwik/src/core/use/use-core.ts

```diff
@@ -123,6 +124,26 @@ export function invoke<ARGS extends any[] = any[], RET = any>(
   return returnValue;
 }
 
+export function asyncInvoke<ARGS extends any[] = any[], RET = any>(
```

Comment:
actually, this isn't possible in the browser environment. Concurrent invokes will mess up things unless you use async localstorage.

Reply:
we can just have a check for now 
```js
                    if (maybeAsync && isServerPlatform())  {
                        return asyncInvoke.call(this, context, fn, ...args);
                    }
                    return invoke.call(this, context, f, ...args);
```

Reply:
I got all this working I had to change a few things

Reply:
@hbendev if you have time right now. Can update your PR I can apply my changes after and we can get this merged

Reply:
Hi @PatrickJS. Just this change you mean? I can surely add in a few minutes.

Reply:
@hbendev yeah fix the conflicts and I'll refactor your pr with my changes

Reply:
On it!